snuba/datasets/processors/metrics_bucket_processor.py

```diff
@@ -35,6 +36,39 @@ def _should_process(self, message: Mapping[str, Any]) -> bool:
     def _process_values(self, message: Mapping[str, Any]) -> Mapping[str, Any]:
         raise NotImplementedError
 
+    def _timeseries_id_token(
+        self,
+        message: Mapping[str, Any],
+        sorted_tag_items: Iterable[Tuple[str, Union[int, str]]],
+    ) -> bytearray:
+        org_id: int = message["org_id"]
+        project_id: int = message["project_id"]
+        metric_id: int = message["metric_id"]
+
+        buffer = bytearray()
+        for field in [org_id, project_id, metric_id]:
+            buffer.extend(field.to_bytes(length=8, byteorder="little"))
+        for (key, value) in sorted_tag_items:
+            buffer.extend(bytes(key, "utf-8"))
+            if isinstance(value, int):
+                buffer.extend(value.to_bytes(length=8, byteorder="little"))
+            elif isinstance(value, str):
+                buffer.extend(bytes(value, "utf-8"))
```

Comment:
Not entirely sure if we need to be checking for both cases, if I understand correctly the metrics indexer takes has the mapping from strings to integer indices

Reply:
There is no harm in adding code to handle the case of tag value being a string. Your understanding is correct that the tag values would be ints right now on the release health metrics pipeline.

Reply:
This code already exists in dynamic sampling. 
Cna we use the same hash and support both the cases where tags are int and the cases where the tags are strings ?
https://github.com/getsentry/snuba/blob/ba2daa237a95da5b4700df1451908859c259b326/snuba/datasets/processors/generic_metrics_processor.py#L49

Reply:
This line: https://github.com/getsentry/snuba/blob/master/snuba/datasets/processors/metrics_bucket_processor.py#L54 suggests that values must be ints, are you suggesting to change this?

Reply:
Are you suggesting to pull out the hash and `_timeseries_id_token` functions outside of the Generic Metrics Processor? I thought it made sense to make it part of each class in case they ever had different implementations. I can pull these functions out, if you don't think that would be a concern.

Reply:
If we had a common implementation, there are more chances of bugs creeping in when a particular dataset would like to change the implementation of timeseries for its own purpose. So duplication should be fine for this.